pkg/plugins/manager/pipeline/bootstrap/steps.go

```diff
@@ -0,0 +1,146 @@
+	"github.com/grafana/grafana/pkg/plugins/log"
+	"github.com/grafana/grafana/pkg/plugins/manager/loader/assetpath"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+// DefaultConstructor implements the default ConstructFunc used for the Construct step of the Bootstrap stage.
+//
+// It uses a pluginFactoryFunc to create plugins and the signatureCalculator to calculate the plugin's signature state.
+type DefaultConstructor struct {
+	pluginFactoryFunc   pluginFactoryFunc
+	signatureCalculator plugins.SignatureCalculator
+	log                 log.Logger
+}
+
+// DefaultConstructFunc is the default ConstructFunc used for the Construct step of the Bootstrap stage.
+var DefaultConstructFunc = func(signatureCalculator plugins.SignatureCalculator, assetPath *assetpath.Service) ConstructFunc {
+	return NewDefaultConstructor(signatureCalculator, assetPath).Construct
+}
+
+// DefaultDecorateFuncs is the default DecorateFuncs used for the Decorate step of the Bootstrap stage.
+var DefaultDecorateFuncs = []DecorateFunc{
+	AliasDecorateFunc,
+	AppDefaultNavURLDecorateFunc,
+	AppChildDecorateFunc,
+}
+
+// NewDefaultConstructor returns a new DefaultConstructor.
+func NewDefaultConstructor(signatureCalculator plugins.SignatureCalculator, assetPath *assetpath.Service) *DefaultConstructor {
+	return &DefaultConstructor{
+		pluginFactoryFunc:   NewDefaultPluginFactory(assetPath).createPlugin,
+		signatureCalculator: signatureCalculator,
+		log:                 log.New("plugins.construct"),
+	}
+}
+
+// Construct will calculate the plugin's signature state and create the plugin using the pluginFactoryFunc.
+func (c *DefaultConstructor) Construct(ctx context.Context, src plugins.PluginSource, bundles []*plugins.FoundBundle) ([]*plugins.Plugin, error) {
+	res := make([]*plugins.Plugin, 0, len(bundles))
+
+	for _, bundle := range bundles {
+		sig, err := c.signatureCalculator.Calculate(ctx, src, bundle.Primary)
+		if err != nil {
+			c.log.Warn("Could not calculate plugin signature state", "pluginID", bundle.Primary.JSONData.ID, "err", err)
+			continue
+		}
+		plugin, err := c.pluginFactoryFunc(bundle.Primary, src.PluginClass(ctx), sig)
+		if err != nil {
+			c.log.Error("Could not create primary plugin base", "pluginID", bundle.Primary.JSONData.ID, "err", err)
+			continue
+		}
+		res = append(res, plugin)
+
+		children := make([]*plugins.Plugin, 0, len(bundle.Children))
+		for _, child := range bundle.Children {
+			cp, err := c.pluginFactoryFunc(*child, plugin.Class, sig)
+			if err != nil {
+				c.log.Error("Could not create child plugin base", "pluginID", child.JSONData.ID, "err", err)
+				continue
+			}
+			cp.Parent = plugin
+			plugin.Children = append(plugin.Children, cp)
+
+			children = append(children, cp)
+		}
+		res = append(res, children...)
+	}
+
+	return res, nil
+}
+
+// AliasDecorateFunc is a DecorateFunc that sets the alias for the plugin.
+var AliasDecorateFunc = DecorateFunc(func(ctx context.Context, p *plugins.Plugin) (*plugins.Plugin, error) {
```

Comment:
Nit: This is just a style thing, so up to you, but I am not a fan of declaring variables like this in the global scope :stuck_out_tongue: .
I would declare those using `func` rather than variables:

```suggestion
func AliasDecorateFunc(ctx context.Context, p *plugins.Plugin) (*plugins.Plugin, error) {
```

Reply:
Interesting 😄  Will change this to see how it feels!